cmd/build.go

```diff
@@ -173,6 +174,50 @@ func GenerateKustomizeResult(config opConfig.Config, kustomizeTemplate template.
 		yamlFile.PutWithSeparator("metalLbSecretKey", base64.StdEncoding.EncodeToString(metalLbSecretKey), ".")
 	}
 
+	artifactRepoS3Node, _ := yamlFile.Get("artifactRepository.s3")
+	if artifactRepoS3Node != nil {
+		_, keyFormat := yamlFile.Get("artifactRepository.s3.keyFormat")
+		_, bucket := yamlFile.Get("artifactRepository.s3.bucket")
+		_, endpoint := yamlFile.Get("artifactRepository.s3.endpoint")
+		_, insecure := yamlFile.Get("artifactRepository.s3.insecure")
+		_, region := yamlFile.Get("artifactRepository.s3.region")
+		_, accessKey := yamlFile.Get("artifactRepository.s3.accessKey")
+		_, secretKey := yamlFile.Get("artifactRepository.s3.secretKey")
+		artifactRepoS3NodeVal := fmt.Sprintf(""+
```

Comment:
Is there a way to do this by mutating the YAML tree instead of a big string like this?

Reply:
Changes are implemented.